src/code_fa4c0_len_3bf0.c

```diff
@@ -22,23 +22,60 @@ ApiStatus SetMusicTrack(ScriptInstance* script, s32 isInitialCall) {
     return (set_music_track(musicPlayer, songID, variation, 0x1F4, volume) != 0) * 2;
 }
 
-INCLUDE_ASM(s32, "code_fa4c0_len_3bf0", FadeInMusic, ScriptInstance* script, s32 isInitialCall);
+ApiStatus FadeInMusic(ScriptInstance* script, s32 isInitialCall) {
+    Bytecode* args = script->ptrReadPos;
+    s32 var0 = get_variable(script, *args++);
+    s32 var1 = get_variable(script, *args++);
+    s32 var2 = get_variable(script, *args++);
+    s32 var3 = get_variable(script, *args++);
+    s16 var4 = get_variable(script, *args++);
+    s16 var5 = get_variable(script, *args++);
+
+    return (func_8014A964(var0, var1, var2, var3, var4, var5) != 0) * 2;
```

Comment:
Can this be expressed with an `if` or a ternary in a matching way? The boolean arithmetic here isn't very fun

Reply:
We have other functions in the codebase that do this. I imagine it's required, as it is a concise way to return either BLOCK or DONE2. I agree it looks kinda weird, but this is the only way I've been able to write these

Reply:
for clarity, could we potentially multiply it by ApiStatus_DONE2 here, so it's clear what's going on? Otherwise, I think it is indeed what the original devs probably wrote.

Reply:
Totally agree, looks way better that way. dun